### PR TITLE
Fix various AC bugs

### PR DESCRIPTION
**Pull request by Atlante45**
[Manuscript Ticket](https://highfidelity.manuscript.com/f/cases/20121/Agents-crash-in-ResourceCaches-on-shutdown)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15549/HighFidelity-Beta-PR14506-c53c52c.apk

**Comment by antoninahifi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15585/HighFidelity-Beta-PR14506-c53c52c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15585/HighFidelity-Beta-Interface-PR14506-c53c52c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15585/HighFidelity-Beta-PR14506-c53c52c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15585/HighFidelity-Beta-Interface-PR14506-c53c52c.exe

**Comment by antoninahifi** [comment]
Currently testing

**Comment by antoninahifi** [comment]
@conklin94122 PASS

File: assignment-client/src/AssignmentClient.cpp [hunk 1]
```diff
@@ -129,17 +129,12 @@ void AssignmentClient::stopAssignmentClient() {
...
-        // when this thread is destroyed we don't need to run our assignment complete method
-        disconnect(currentAssignmentThread, &QThread::destroyed, this, &AssignmentClient::assignmentCompleted);
-
-        // wait on the thread from that assignment - it will be gone once the current assignment deletes
-        currentAssignmentThread->quit();
-        currentAssignmentThread->wait();
+        auto PROCESS_EVENTS_INTERVAL_MS = 100;
+        while (!currentAssignmentThread->wait(PROCESS_EVENTS_INTERVAL_MS)) {
+            QCoreApplication::processEvents();
+        }
```
Comment (SimonWalton-HiFi) [inline assignment-client/src/AssignmentClient.cpp]: Does the ThreadedAssignment still get deleted for all AC types?

Comment (Atlante45) [reply assignment-client/src/AssignmentClient.cpp]: Yes, this code is generic to all ACs.
We were actually calling `deleteLater` twice.
All the quit/delete logic is setup in `handleCreateAssignmentPacket`

**Comment by Atlante45** [comment]
@simon-walton You're correct, I should really rename this `AC bugs`. I noticed the same shutdown cleanup issues that were in the Agent were also present in the ESS.

**Pull request merged**